testing/integration/nginx_unprivileged_test.go

```diff
@@ -0,0 +1,164 @@
+	healthCheckTime        time.Duration
+	healthCheckRefreshTime time.Duration
+}
+
+func TestNginxUnprivileged(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Group: "fleet",
+		Stack: &define.Stack{},
+		OS: []define.OS{
+			{Type: define.Linux},
+		},
+		Local: false, // requires Agent installation
+		Sudo:  true,  // requires Agent installation
+	})
+
+	runner := &NginxUnprivilegedRunner{
+		info:                   info,
+		healthCheckTime:        time.Minute * 5,
+		healthCheckRefreshTime: time.Second * 5,
+	}
+
+	suite.Run(t, runner)
+}
+
+func (runner *NginxUnprivilegedRunner) SetupSuite() {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+	cmd := exec.CommandContext(ctx, "apt-get", "install", "-v", "nginx-full")
+	out, err := cmd.CombinedOutput()
+	require.NoError(runner.T(), err, "error while installing nginx: %s", string(out))
+
+	err = os.WriteFile("/etc/nginx/conf.d/status.conf", []byte(nginxStatusModule), 0750)
+	require.NoError(runner.T(), err)
+
+	cmd = exec.CommandContext(ctx, "nginx", "-s", "reload")
+	out, err = cmd.CombinedOutput()
+	require.NoError(runner.T(), err, "error while reloading nginx: %s", string(out))
+
+	fixture, err := define.NewFixtureFromLocalBuild(runner.T(), define.Version())
+	require.NoError(runner.T(), err)
+	runner.agentFixture = fixture
+
+	policyUUID := uuid.Must(uuid.NewV4()).String()
+	basePolicy := kibana.AgentPolicy{
+		Name:        "test-policy-" + policyUUID,
+		Namespace:   "default",
+		Description: "Test policy " + policyUUID,
+		MonitoringEnabled: []kibana.MonitoringEnabledOption{
+			kibana.MonitoringEnabledLogs,
+			kibana.MonitoringEnabledMetrics,
+		},
+	}
+
+	installOpts := atesting.InstallOpts{
+		NonInteractive: true,
+		Force:          true,
+		Privileged:     false,
+	}
+
+	policyResp, err := tools.InstallAgentWithPolicy(ctx, runner.T(), installOpts, runner.agentFixture, runner.info.KibanaClient, basePolicy)
+	require.NoError(runner.T(), err)
+
+	runner.policyID = policyResp.ID
+	runner.policyName = basePolicy.Name
+
+	_, err = tools.InstallPackageFromDefaultFile(ctx, runner.info.KibanaClient, "nginx", "1.23.0", "nginx_integration_setup.json", uuid.Must(uuid.NewV4()).String(), policyResp.ID)
+	require.NoError(runner.T(), err)
+}
+
+func (runner *NginxUnprivilegedRunner) TestComponentHealth() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*10)
+	defer cancel()
+
+	runner.AllComponentsHealthy(ctx)
+
+}
+
+// AllComponentsHealthy ensures all the beats and agent are healthy and working before we continue
+func (runner *NginxUnprivilegedRunner) AllComponentsHealthy(ctx context.Context) {
+	compDebugName := ""
+	require.Eventually(runner.T(), func() bool {
+		allHealthy := true
+		status, err := runner.agentFixture.ExecStatus(ctx)
+		if err != nil {
+			runner.T().Logf("agent status returned an error: %v", err)
+			return false
+		}
+
+		for _, comp := range status.Components {
```

Comment:
Possible to be healthy without any components, should check that components list is not empty and if it is mark it not healthy.

Reply:
It's also possible to be healthy while not yet having read any documents that would require accessing potentially privileged sources.

Just checking health doesn't actually guarantee we tested anything. You need to make sure we ingested data.

Potentially this entire test could be done in https://github.com/elastic/integrations/tree/main/packages/nginx, as long as we have a way to start unprivileged containers. The new Wolfi container would satisfy this by default. 

https://github.com/elastic/integrations/pull/10969 makes the switch to Wolfi. We may get significantly better test coverage by adding this health check to the end of every integration system test, with the advantage that we also won't be exposed to problems in the nginx integration here.

Reply:
> Just checking health doesn't actually guarantee we tested anything. You need to make sure we ingested data.

Sure. Let me add that step.

Reply:
Yeah I don't think we should have this test in our repository. I think this is ultimately a test for github.com/elastic/integrations repository.